src/toast/__tests__/toast.test.js

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import {mount} from 'enzyme';
+import {Toast, StyledBody, StyledCloseIcon, KIND} from '../index';
+
+jest.useFakeTimers();
+
+describe('Toast', () => {
+  test('basic functionality', () => {
+    const wrapper = mount(<Toast>Notification</Toast>);
+
+    expect(wrapper.instance().state.isHidden).toBe(false);
+    expect(wrapper.instance().state.isAnimating).toBe(true);
+
+    expect(wrapper.find(StyledBody).first()).toExist();
+    expect(wrapper.find(StyledCloseIcon).first()).toExist();
+  });
+
+  test('Body component props', () => {
+    let renderedRoot;
+    const wrapper = mount(<Toast>Notification</Toast>);
+
+    renderedRoot = wrapper.find(StyledBody).first();
+    expect(renderedRoot).toExist();
+    expect(renderedRoot.props().$kind).toBe(KIND.info);
```

Comment:
you can use https://jestjs.io/docs/en/expect#tomatchobjectobject to check a subset of object

Reply:
Here I want specifically check the `$kind` prop. I think breaking down the check works well and easy to read.

Reply:
yes. However, if we would like to emphasize on particular logic, I would suggest to do it like this:
```js
  describe('Body component', () => {
     let renderedRoot;
     beforeEach(() => {
         const wrapper = mount(<Toast>Notification</Toast>);
        renderedRoot = wrapper.find(StyledBody).first();
     });
    test('should render Body of Toast immediately', () => {
       expect(renderedRoot).toExist();
    });

    test('should set styles of Toast Body to be of information by default', () => {
       expect(renderedRoot.props().$kind).toBe(KIND.info);
    })
```

Reply:
I think it's a good approach to write tests in the way you mentioned. Currently, we don't follow a consistent pattern in how we structure our unit tests and tests' code. How about opening a new discussion on the unit test writing best practices in order to get feedback from others and try to come to an agreement on the consistency of our unit tests across the library? @blackswanny Would you mind starting this discussion?

Reply:
Sure, thanks. I can provide some slides and ideas on the next Base UI meeting